lib/base_service.ts

```
@@ -192,18 +191,42 @@ export class BaseService {
     }
   }
 
+  /**
+   * Guarantee that the next request you make will be IAM authenticated. This
+   * performs any requests necessary to get a valid IAM token so that if your
+   * next request involves a streaming operation, it will not be interrupted.
+   *
+   * @param {Function} callback - callback function to return flow of execution
+   *
+   * @returns {void}
+   */
+  protected preAuthenticate(callback): void {
+     if (Boolean(this.tokenManager)) {
+      return this.tokenManager.getToken((err, token) => {
+        if (err) {
+          callback();
```

the callback should bubble up the error

```js
return callback(err);
```